addon/object/index.js

```diff
@@ -74,82 +65,86 @@ export const action = decorator(function(target, key, desc) {
- *   var someKey = this.get('someKey');
- *   var otherKey = this.get('otherKey');
+ * import EmberObject from '@ember/object';
+ * import computed from 'ember-decorators/object';
  *
- *   // Do Stuff
- * })
- * ```
+ * export default class extends EmberObject {
+ *   someKey = 'foo';
+ *   otherKey = 'bar';
  *
- * You replace with this:
+ *   person = {
+ *     firstName: 'John',
+ *     lastName: 'Smith'
+ *   };
  *
- * ```javascript
- * import computed from 'ember-decorators/object';
+ *   @computed('someKey', 'otherKey')
+ *   foo(someKey, otherKey) {
+ *     return `${someKey} - ${otherKey}`; // => 'foo - bar'
+ *   }
  *
- * // ..... <snip> .....
- * @computed('someKey', 'otherKey')
- * foo(someKey, otherKey) {
- *   // Do Stuff
+ *   @computed('person.{firstName,lastName}')
+ *   fullName(firstName, lastName) {
+ *     return `${firstName} ${lastName}`; // => 'John Smith'
+ *   }
  * }
  * ```
  *
  * #### Without Dependent Keys
  *
- * ```javascript
- * foo: Ember.computed(function() {
- *   // Do Stuff
- * })
- * ```
- *
- * You replace with this:
+ * Computed properties without dependent keys are cached for the whole life span
+ * of the object. You can only force a recomputation by calling
+ * [`notifyPropertyChange`](https://www.emberjs.com/api/ember/2.14/classes/Ember.Observable/methods/notifyPropertyChange?anchor=notifyPropertyChange)
+ * on the computed property.
  *
  * ```javascript
+ * import EmberObject from '@ember/object';
  * import computed from 'ember-decorators/object';
  *
- * // ..... <snip> .....
- * @computed
- * foo() {
- *   // Do Stuff
+ * export default class extends EmberObject {
+ *   @computed
+ *   foo() {
+ *     // calculate stuff
+ *     return stuff;
+ *   }
  * }
  * ```
  *
- * #### "Real World"
+ * #### Getter and Setter
  *
  * ```javascript
- * import Ember from 'ember';
+ * import Component from '@ember/component';
+ * import { setProperties } from ''@ember/object';
  * import computed from 'ember-decorators/object';
  *
- * export default Ember.Component.extend({
- *   @computed('first', 'last')
- *   name(first, last) {
- *     return `${first} ${last}`;
- *   }
- * });
- * ```
- *
- *
- * #### "Real World get/set syntax"
+ * export default class extends Component {
+ *   first = 'John';
+ *   last = 'Smith';
  *
- * ```javascript
- * import Ember from 'ember';
- * import computed from 'ember-decorators/object';
- *
- * export default Ember.Component.extend({
  *   @computed('first', 'last')
- *   name: {
+ *   name = {
  *     get(first, last) {
- *       return `${first} ${last}`;
+ *       return `${first} ${last}`; // => 'John Smith'
  *     },
  *
  *     set(value, first, last) {
- *       // ...
+ *       if (typeof value !== 'string' || !value.test(/^[a-z]+ [a-z]+$/i)) {
+ *         throw new TypeError('Invalid name');
+ *       }
+ *
+ *       const [first, last] = value.split(' ');
+ *       setProperties(this, { first, last });
+ *
+ *       return value;
```

Comment:
Good catch.

Since `emberComputed` is [`Ember.computed`](https://emberjs.com/api/ember/2.14/namespaces/Ember.computed), I would argue that this is broke at the moment.